src/transform/FooterMenu.js

```diff
@@ -0,0 +1,215 @@
+import './FooterMenu.css'
+
+/**
+ * @typedef {function} FooterMenuItemPayloadExtractor
+ * @param {!Document} document
+ * @return {!string[]} Important - should return empty array if no payload strings.
+ */
+
+/**
+  * @typedef {function} FooterMenuItemClickCallback
+  * @param  {!string[]} payload Important - should return empty array if no payload strings.
+  * @return {void}
+  */
+
+/**
+ * Extracts array of no-html page issues strings from document.
+ * @type {FooterMenuItemPayloadExtractor}
+ */
+const pageIssuesStringsArray = document => {
+  const tables = document.querySelectorAll(
+    'div#content_block_0 table.ambox:not(.ambox-multiple_issues):not(.ambox-notice)'
+  )
+  // Get the tables into a fragment so we can remove some elements without triggering a layout
+  const fragment = document.createDocumentFragment()
+  for (let i = 0; i < tables.length; i++) {
+    fragment.appendChild(tables[i].cloneNode(true))
+  }
+  // Remove some element so their text doesn't appear when we use "innerText"
+  Array.from(fragment.querySelectorAll('.hide-when-compact, .collapsed')).forEach(el => el.remove())
+  // Get the innerText
+  return Array.from(fragment.querySelectorAll('td[class$=mbox-text]')).map(el => el.innerText)
+}
+
+/**
+ * Extracts array of disambiguation page urls from document.
+ * @type {FooterMenuItemPayloadExtractor}
+ */
+const disambiguationTitlesArray = document => Array.from(
+  document.querySelectorAll(
+    'div#content_block_0 div.hatnote a[href]:not([href=""]):not([redlink="1"])'
+  )
+).map(el => el.href)
+
+/**
+ * Type representing kinds of menu items.
+ * @type {Object}
```

Comment:
I'm still fuzzy on how these `@type` / `@typedef` are intended to work but I think this should be `@type {!MenuItemType}` and in another comment you'd do `@typedef {{name: !string, id: !number}} MenuItemType` or

```
@typedef {!object} MenuItemType
@prop {!number} languages
@prop {!number} lastEdited
```